BUMP_VERSION.asciidoc

```diff
@@ -0,0 +1,196 @@
+be generated automatically.
+-------------------------------------
+  -bwcVersion = '5.2.0-SNAPSHOT'
+  +bwcVersion = '5.3.0-SNAPSHOT'
+-------------------------------------
+
+Before the rolling upgrade tests can be run, however, make sure the 5.3.0-SNAPSHOT build is already
+published to the staging artifacts (there is an internal CI job to generate the 5.x snapshot build).
+This should also go away soon, as work is underway to not require pulling down a snapshot build of
+the backward compatibility version but instead, the rolling upgrade test will checkout the backward
+compatibility version in question to a git worktree.
+
+
+== Steps to take after releasing
+
+After releasing a version of Elasticsearch (in our scenario, 5.2.0), then the following must 
+be done to bump to the next version of Elasticsearch.  Note that the same procedure is followed
+when bumping the bug-fix version (for example, after releasing 5.2.1, bumping the release version
+to 5.2.2).
+
+==== Upgrade the version
+
+Since 5.2.0 has been released, the next bug-fix version in the 5.2.0 line 
+will be 5.2.1.  Create a V_5_2_1 version constant in `core/src/main/java/org/elasticsearch/Version.java`
+for all branches >= 5.2.  In this case, the V_5_2_1 version constant should go into
+the 5.2, 5.x, and master branches.  
+
+For the 5.2 branch only, set the CURRENT version in `Version.java` to the new V_5_2_1 constant.  
+For the 5.x and master branches, just add the V_5_2_1 constant without setting the CURRENT version.
+-------------------------------------
+  + public static final int V_5_2_1_ID_UNRELEASED = 5020199;
+  + public static final Version V_5_2_1_UNRELEASED = new Version(V_5_2_1_ID, org.apache.lucene.util.Version.LUCENE_6_3_0);
+  + public static final Version CURRENT = V_5_2_1_UNRELEASED;
+-------------------------------------
+
+Also, update the `Version#fromId` method to include the new 5.2.1 version in each branch
+-------------------------------------
+   public static Version fromId(int id) {
+       switch (id) {
+  +       case V_5_2_1_ID_UNRELEASED:
+  +           return V_5_2_1_UNRELEASED;
+-------------------------------------
+
+==== Upgrade version.properties
+
+In the 5.2 branch, change the elasticsearch version in `buildSrc/version.properties` to 5.2.1:
+-------------------------------------
+  -elasticsearch     = 5.2.0
+  +elasticsearch     = 5.2.1
+-------------------------------------
+
+==== Add the version to the vagrant tests
+
+In the `qa/vagrant/versions` file, add the newly released 5.2.0:
+-------------------------------------
+  +5.2.0
+-------------------------------------
+
+==== Generate the backward compatibility indices
+
+Generate the backward compatibility (bwc) indices for the bwc tests for the just released 5.2.0 
+with the following steps:
+
+- Make sure python3 is installed, if you don't already have it (on a Mac, execute `brew install python3`)
```

Comment:
Please recommend using a virtual environment instead of installing packages on the system:

```bash
$ python3 -m venv <path to virtual environment>
```

and then

```bash
$ <path to virtual environment>/bin/pip3 install requests
```

and

```bash
$ <path to virtual envrionment>/bin/python3 ./create_bwc_index.py 5.2.0
```

for example (and so on).